### ISSUE TITLE
[BUG] Playback of 44.1 kHz content via SRC pipe crashes firmware

### ISSUE THREAD

**Issue by singalsu** [thesofproject/sof#1236]
**Describe the bug**
Firmware crash happens in idle after a number of plays of 44.1 kHz sample rate stereo clips. The word length of clip or the length of the clip does not impact crash probability. Quite often third play ends into a crash. Playback of other sample rates 8/16/32/48 kHz do not crash (8 -> 48 kHz is using same SRC two stage structure as 44.1 -> 48). Also xt-xcc built FW does not crash, not even if compiling the same source as for gcc without intrinsic optimizatin.

**To Reproduce**
In UP2 use topology sof-apl-src-pcm512x.tplg.
Then repeat command "aplay -Dhw:0,0 song.wav"
You can see open kernel log into another ssh window with "journalctl -f -k". Observe also sof logger output.

**Expected behavior**
Nomal SOF idle after playback.

**Impact**
This is a showstopper bug because the device need to be rebooted to recover.

**Environment**
1) Branch name and commit hash of 3 repositories: sof (firmware), linux (kernel driver) and soft (tools & topology): Git master firmware, topic/sof-dev kernel.
2) Name of the topology file: sof-apl-src-pcm512x.tplg  
3) Name of the platform(s) on which the bug is observed: UP2
4) Reproducibility Rate. If you can only reproduce it randomly, it’s useful to report how many times the bug has been reproduced vs. the number of attempts it’s taken to reproduce the bug: Usually third playback ends to a crash.

**Screenshots or console output**
If applicable, add a screenshot (drag-and-drop an image), or console logs
(cut-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue.

Please also include the relevant sections from the firmware log and kernel log in the report (and attach the full logs for complete reference). Kernel log is taken from *dmesg* and firmware log from *sof-logger*. See https://thesofproject.github.io/latest/developer_guides/debugability/logger/index.html

dmesg

```
[ 3509.204215] sof-audio-pci 0000:00:0e.0: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[ 3509.204387] sof-audio-pci 0000:00:0e.0: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[ 3510.175161] sof-audio-pci 0000:00:0e.0: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 3510.175175] sof-audio-pci 0000:00:0e.0: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 3542.231166] sof-audio-pci 0000:00:0e.0: pcm: trigger stream 0 dir 0 cmd 0
[ 3542.231183] sof-audio-pci 0000:00:0e.0: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[ 3542.231436] sof-audio-pci 0000:00:0e.0: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[ 3542.231780] sof-audio-pci 0000:00:0e.0: pcm: free stream 0 dir 0
[ 3542.231792] sof-audio-pci 0000:00:0e.0: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 3542.231931] sof-audio-pci 0000:00:0e.0: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 3542.232067] sof-audio-pci 0000:00:0e.0: error : DSP panic!
[ 3542.232080] sof-audio-pci 0000:00:0e.0: panic: dsp_oops_offset 788480 offset 788480
[ 3542.232088] sof-audio-pci 0000:00:0e.0: status: fw entered - code 00000005
[ 3542.232132] sof-audio-pci 0000:00:0e.0: error: runtime exception
[ 3542.232139] sof-audio-pci 0000:00:0e.0: error: trace point 00004000
[ 3542.232145] sof-audio-pci 0000:00:0e.0: error: panic happen at :0
[ 3542.232152] sof-audio-pci 0000:00:0e.0: error: DSP Firmware Oops
[ 3542.232158] sof-audio-pci 0000:00:0e.0: error: Exception Cause: IllegalInstructionCause, Illegal instruction
[ 3542.232166] sof-audio-pci 0000:00:0e.0: EXCCAUSE 0x00000000 EXCVADDR 0x00000000 PS       0x00000000 SAR     0x00000000
[ 3542.232174] sof-audio-pci 0000:00:0e.0: EPC1     0x00000000 EPC2     0x00000000 EPC3     0x00000000 EPC4    0x00000000
[ 3542.232181] sof-audio-pci 0000:00:0e.0: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[ 3542.232188] sof-audio-pci 0000:00:0e.0: EPS2     0x00000000 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
[ 3542.232195] sof-audio-pci 0000:00:0e.0: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000000
[ 3542.232201] sof-audio-pci 0000:00:0e.0: stack dump from 0x00000000
[ 3542.232212] sof-audio-pci 0000:00:0e.0: 0x00000000: 00000000 00000000 00000000 00000000
[ 3542.232220] sof-audio-pci 0000:00:0e.0: 0x00000004: 00000000 00000000 00000000 00000000
[ 3542.232229] sof-audio-pci 0000:00:0e.0: 0x00000008: 600f3800 27906847 f8d00028 ffff94a8
[ 3542.232237] sof-audio-pci 0000:00:0e.0: 0x0000000c: f9467880 ffff94a8 f7db9400 ffff94a8
[ 3542.232246] sof-audio-pci 0000:00:0e.0: 0x00000010: a18b4772 ffffffff 00000000 00000000
[ 3542.232254] sof-audio-pci 0000:00:0e.0: 0x00000014: fa408d00 ffff94a8 f9467880 ffff94a8
[ 3542.232263] sof-audio-pci 0000:00:0e.0: 0x00000018: 00000000 00000000 00000000 00000000
[ 3542.232271] sof-audio-pci 0000:00:0e.0: 0x0000001c: 00000000 00000000 00000000 00000000
[ 3542.232277] sof-audio-pci 0000:00:0e.0: error: waking up any trace sleepers
[ 3542.232409] sof-audio-pci 0000:00:0e.0: error: trace IO error
[ 3542.245463] sof-audio-pci 0000:00:0e.0: pcm: close stream 0 dir 0
[ 3543.640627] sof-audio-pci 0000:00:0e.0: error : DSP panic!
[ 3543.640643] sof-audio-pci 0000:00:0e.0: panic: dsp_oops_offset 788480 offset 788480

```

Sof-logger error trace (normal trace doesn't show this)

```
 CORE  LEVEL      COMP_ID                TIMESTAMP            DELTA                FILE_NAME	CONTENT
    0      2           SA          34500421.510417  34500420.000000      src/lib/agent.c:74  	validate(), idle longer than timeout, delta = 27060583
    0      2         PIPE 1.7      33091264.635417              nan src/audio/pipeline.c:533 	pipeline_trigger()
    0      2         HOST          33091268.229167         3.593750     src/audio/host.c:281 	host_trigger()
    0      2          DMA          33091271.927083         3.697917 intel/cavs/hda-dma.c:645 	hda-dmac: 5 channel 0 -> stop
    0      2          SRC          33091275.625000         3.697917      src/audio/src.c:713 	src_trigger()
    0      2       VOLUME          33091278.906250         3.281250   src/audio/volume.c:454 	volume_trigger()
    0      2          DAI 1.6      33091282.395833         3.489583      src/audio/dai.c:544 	dai_comp_trigger(), command = 0
    0      2          DAI 1.6      33091285.625000         3.229167      src/audio/dai.c:608 	dai_comp_trigger(), PAUSE/STOP
    0      2          DMA          33091288.906250         3.281250 src/drivers/dw/dma.c:675 	dw-dma: 0 channel 0 stop
    0      2          SSP          33091295.677083         6.770833 ers/intel/cavs/ssp.c:812 	ssp_trigger() cmd 0
    0      2          SSP          33091425.729167       130.052078 ers/intel/cavs/ssp.c:793 	ssp_stop(), TX stop
    0      2          SSP          33091428.958333         3.229167 ers/intel/cavs/ssp.c:802 	ssp_stop(), SSP port disabled
    0      2          IPC          33091868.541667       439.583344    src/ipc/handler.c:367 	ipc: comp 0 -> free
    0      2         PIPE 1.7      33091872.291667         3.750000 src/audio/pipeline.c:576 	pipeline_reset()
    0      2         HOST          33091875.468750         3.177083     src/audio/host.c:704 	host_reset()
    0      2          DMA          33091878.854167         3.385417 intel/cavs/hda-dma.c:645 	hda-dmac: 5 channel 0 -> stop
    0      2          SRC          33091884.739583         5.885417      src/audio/src.c:925 	src_reset()
    0      2       VOLUME          33091888.125000         3.385417   src/audio/volume.c:621 	volume_reset()
    0      2          DAI 1.6      33091891.562500         3.437500      src/audio/dai.c:521 	dai_reset()

```

**Comment by singalsu** [comment]
This issue was fixed by PR #1267 (commit 2e9264586b8d756b393b519748b552187c8339aa). I just completed playback of 10 clips with 44.1 kHz rate and no problems. Thanks @tlauda !

**Comment by singalsu** [comment]
Closing!
